src/dune_engine/build_config.mli

```diff
@@ -33,10 +33,14 @@ type rules =
   ; rules : Rules.t Memo.t
   }
 
+val empty_rules : rules
+
+val combine_rules : rules -> rules -> rules
```

Comment:
How about exporting the `Monoid` interface that we have in Stdune instead? Then we might be able to reuse some functions.

Reply:
Hmm, actually, I guess it's not a monoid because the operation is partial. And, I suppose `combine_rules` is missing the `_exn` suffix to indicate that it can raise?

Reply:
(Also, let's put this functionality into a separate submodule `Rules` maybe?)